Ubuntu main: 4 tests fail

### Bug description

Four unit tests fail on main branch with Ubuntu.

### Configuration

_No response_

### Command used

```shell
pytest
```


### Pylint output

```shell
edreamleo@edreamleo-MS-7D09:~$ cd ekr-pylint/
edreamleo@edreamleo-MS-7D09:~/ekr-pylint$ pytest
============================= test session starts ==============================
platform linux -- Python 3.10.4, pytest-7.1.2, pluggy-1.0.0
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /home/edreamleo/ekr-pylint, configfile: setup.cfg, testpaths: tests
plugins: benchmark-3.4.1
collected 1972 items                                                           

tests/test_check_parallel.py .s.....sssssssssssssssssssssssssss          [  1%]
tests/test_deprecation.py .......                                        [  2%]
tests/test_epylint.py ..                                                 [  2%]
tests/test_func.py ............                                          [  2%]
tests/test_functional.py ............................................... [  5%]
........................................................................ [  8%]
.....s..........s.....................................................s. [ 12%]
........................................................................ [ 16%]
....................s.s.s.......................s..............s........ [ 19%]
...................................................................s...s [ 23%]
..F..........s...........ss...s.s....................................... [ 27%]
........F..F.F..................................................s...s... [ 30%]
.......................................................s................ [ 34%]
.....................................s.....s............................ [ 38%]
.................................................................                                                                                                                                       [ 41%]
tests/test_functional_directories.py .                                                                                                                                                                  [ 41%]
tests/test_import_graph.py ......                                                                                                                                                                       [ 41%]
tests/test_numversion.py ...........                                                                                                                                                                    [ 42%]
tests/test_pragma_parser.py ............                                                                                                                                                                [ 42%]
tests/test_pylint_runners.py ........                                                                                                                                                                   [ 43%]
tests/test_regr.py .................x.                                                                                                                                                                  [ 44%]
tests/test_self.py ..........x.......................................... [ 46%]
.............s.....................................................                                                                                                                                     [ 50%]
tests/test_similar.py ............                                                                                                                                                                      [ 50%]
tests/unittest_reporters_json.py ..                                                                                                                                                                     [ 51%]
tests/unittest_reporting.py .........                                                                                                                                                                   [ 51%]
tests/benchmark/test_baseline_benchmarks.py .ss.s.s.s..                                                                                                                                                 [ 52%]
tests/checkers/unittest_base_checker.py ..                                                                                                                                                              [ 52%]
tests/checkers/unittest_deprecated.py ........................                                                                                                                                          [ 53%]
tests/checkers/unittest_design.py ...                                                                                                                                                                   [ 53%]
tests/checkers/unittest_format.py .........                                                                                                                                                             [ 53%]
tests/checkers/unittest_imports.py ......                                                                                                                                                               [ 54%]
tests/checkers/unittest_misc.py ...........                                                                                                                                                             [ 54%]
tests/checkers/unittest_non_ascii_name.py ..........................x                                                                                                                                   [ 56%]
tests/checkers/unittest_refactoring.py ..                                                                                                                                                               [ 56%]
tests/checkers/unittest_similar.py ..................                                                                                                                                                   [ 57%]
tests/checkers/unittest_spelling.py ssssssssssssssssssssssssssssssssss                                                                                                                                  [ 58%]
tests/checkers/unittest_stdlib.py .                                                                                                                                                                     [ 58%]
tests/checkers/unittest_strings.py .                                                                                                                                                                    [ 59%]
tests/checkers/unittest_typecheck.py ss.                                                                                                                                                                [ 59%]
tests/checkers/unittest_utils.py ....................................... [ 61%]
                                                                                                                                                                                                        [ 61%]
tests/checkers/unittest_variables.py ..........                                                                                                                                                         [ 61%]
tests/checkers/base/unittest_base.py s                                                                                                                                                                  [ 61%]
tests/checkers/base/unittest_multi_naming_style.py ....                                                                                                                                                 [ 61%]
tests/checkers/base/unittest_name_preset.py ....                                                                                                                                                        [ 62%]
tests/checkers/unittest_unicode/unittest_bad_chars.py .................. [ 63%]
..................................                                                                                                                                                                      [ 64%]
tests/checkers/unittest_unicode/unittest_bidirectional_unicode.py ...... [ 65%]
.............................................                                                                                                                                                           [ 67%]
tests/checkers/unittest_unicode/unittest_functions.py .................. [ 68%]
........................................................................ [ 71%]
..                                                                                                                                                                                                      [ 72%]
tests/checkers/unittest_unicode/unittest_invalid_encoding.py xx........x [ 72%]
...........                                                                                                                                                                                             [ 73%]
tests/config/test_argparse_config.py ......                                                                                                                                                             [ 73%]
tests/config/test_config.py .........                                                                                                                                                                   [ 73%]
tests/config/test_deprecations.py ............                                                                                                                                                          [ 74%]
tests/config/test_find_default_config_files.py ............                                                                                                                                             [ 75%]
tests/config/test_functional_config_loading.py ........................                                                                                                                                 [ 76%]
tests/config/unittest_config.py ........                                                                                                                                                                [ 76%]
tests/extensions/test_check_docs_utils.py ..............                                                                                                                                                [ 77%]
tests/extensions/test_private_import.py .....                                                                                                                                                           [ 77%]
tests/functional/r/redundant_unittest_assert.py ss                                                                                                                                                      [ 77%]
tests/lint/test_caching.py .....ss....                                                                                                                                                                  [ 78%]
tests/lint/test_pylinter.py ..                                                                                                                                                                          [ 78%]
tests/lint/test_utils.py ...                                                                                                                                                                            [ 78%]
tests/lint/unittest_expand_modules.py .....                                                                                                                                                             [ 78%]
tests/lint/unittest_lint.py ...........................................s [ 81%]
ss...                                                                                                                                                                                                   [ 81%]
tests/message/test_no_removed_msgid_or_symbol_used.py .                                                                                                                                                 [ 81%]
tests/message/unittest_message.py .                                                                                                                                                                     [ 81%]
tests/message/unittest_message_definition.py ......                                                                                                                                                     [ 81%]
tests/message/unittest_message_definition_store.py .................                                                                                                                                    [ 82%]
tests/message/unittest_message_id_store.py ........                                                                                                                                                     [ 83%]
tests/primer/test_primer_external.py sssssssssss                                                                                                                                                        [ 83%]
tests/primer/test_primer_stdlib.py sssssssssssssssssssssssssssssssssssss [ 85%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 89%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 92%]
sssssssssssssssssssss                                                                                                                                                                                   [ 93%]
tests/profile/test_profile_against_externals.py s                                                                                                                                                       [ 93%]
tests/pyreverse/test_diadefs.py .........                                                                                                                                                               [ 94%]
tests/pyreverse/test_diagrams.py .                                                                                                                                                                      [ 94%]
tests/pyreverse/test_inspector.py ........                                                                                                                                                              [ 94%]
tests/pyreverse/test_main.py .....................                                                                                                                                                      [ 95%]
tests/pyreverse/test_printer.py ..............                                                                                                                                                          [ 96%]
tests/pyreverse/test_printer_factory.py .....                                                                                                                                                           [ 96%]
tests/pyreverse/test_pyreverse_functional.py ...                                                                                                                                                        [ 96%]
tests/pyreverse/test_utils.py ......................                                                                                                                                                    [ 98%]
tests/pyreverse/test_writer.py ...............                                                                                                                                                          [ 98%]
tests/testutils/test_configuration_test.py .                                                                                                                                                            [ 98%]
tests/testutils/test_functional_testutils.py ....                                                                                                                                                       [ 99%]
tests/testutils/test_lint_module_output_update.py s...                                                                                                                                                  [ 99%]
tests/testutils/test_output_line.py ........                                                                                                                                                            [ 99%]
tests/testutils/test_package_to_lint.py ..                                                                                                                                                              [ 99%]
tests/utils/unittest_ast_walker.py ..                                                                                                                                                                   [ 99%]
tests/utils/unittest_utils.py ..                                                                                                                                                                        [100%]

================================================================================================== FAILURES ===================================================================================================
___________________________________________________________________________ test_functional[super_init_not_called_extensions_py310] ___________________________________________________________________________

self = <pylint.testutils.lint_module_test.LintModuleTest object at 0x7f0e05fe2a40>

    def runTest(self) -> None:
>       self._runTest()
E       AssertionError: Wrong results for file "super_init_not_called_extensions_py310":
E       
E       Unexpected in testdata:
E          4: import-error
E         14: super-init-not-called

pylint/testutils/lint_module_test.py:115: AssertionError
-------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------
************* Module /home/edreamleo/ekr-pylint/tests/functional/s/super/super_init_not_called_extensions_py310.rc
tests/functional/s/super/super_init_not_called_extensions_py310.rc:1:0: E0015: Unrecognized option found: exclude_platforms, min_pyver (unrecognized-option)
________________________________________________________________________________ test_functional[typing_broken_noreturn_py372] ________________________________________________________________________________

self = <pylint.testutils.lint_module_test.LintModuleTest object at 0x7f0e04550f10>

    def runTest(self) -> None:
>       self._runTest()
E       AssertionError: Wrong results for file "typing_broken_noreturn_py372":
E       
E       Unexpected in testdata:
E         13: import-error

pylint/testutils/lint_module_test.py:115: AssertionError
____________________________________________________________________________ test_functional[typing_broken_noreturn_future_import] ____________________________________________________________________________

self = <pylint.testutils.lint_module_test.LintModuleTest object at 0x7f0e043ff430>

    def runTest(self) -> None:
>       self._runTest()
E       AssertionError: Wrong results for file "typing_broken_noreturn_future_import":
E       
E       Unexpected in testdata:
E         16: import-error

pylint/testutils/lint_module_test.py:115: AssertionError
-------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------
************* Module /home/edreamleo/ekr-pylint/tests/functional/ext/typing/typing_broken_noreturn_future_import.rc
tests/functional/ext/typing/typing_broken_noreturn_future_import.rc:1:0: E0015: Unrecognized option found: min_pyver (unrecognized-option)
___________________________________________________________________________________ test_functional[typing_broken_noreturn] ___________________________________________________________________________________

self = <pylint.testutils.lint_module_test.LintModuleTest object at 0x7f0e043a78b0>

    def runTest(self) -> None:
>       self._runTest()
E       AssertionError: Wrong results for file "typing_broken_noreturn":
E       
E       Expected in testdata:
E         26: broken-noreturn
E       
E       Unexpected in testdata:
E         11: import-error

pylint/testutils/lint_module_test.py:115: AssertionError

--------------------------------------------------------------------------------------------------------------- benchmark 'baseline': 6 tests ----------------------------------------------------------------------------------------------------------------
Name (time in us)                                                  Min                       Max                      Mean                StdDev                    Median                   IQR            Outliers         OPS            Rounds  Iterations
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_baseline_benchmark_j1                                    242.5650 (1.0)            350.4700 (1.0)            248.3038 (1.0)          4.7851 (1.0)            247.5360 (1.0)          3.3775 (1.0)        163;93  4,027.3246 (1.0)        2033           1
test_baseline_benchmark_j1_all_checks_single_file          23,742.7890 (97.88)       24,792.5620 (70.74)       24,103.2669 (97.07)      281.5767 (58.84)       24,118.7385 (97.44)      362.5540 (107.34)       13;2     41.4882 (0.01)         38           1
test_baseline_lots_of_files_j1                             77,316.6520 (318.75)      77,888.1830 (222.24)      77,544.7623 (312.30)     185.5564 (38.78)       77,507.4370 (313.12)     305.0265 (90.31)         6;0     12.8958 (0.00)         13           1
test_baseline_lots_of_files_j1_empty_checker               77,350.3950 (318.89)      77,933.6830 (222.37)      77,593.0744 (312.49)     155.3858 (32.47)       77,555.4780 (313.31)     186.6175 (55.25)         4;0     12.8877 (0.00)         13           1
test_baseline_benchmark_j1_all_checks_lots_of_files       477,223.9660 (>1000.0)    484,824.5590 (>1000.0)    481,469.9888 (>1000.0)  3,436.2339 (718.11)     483,242.3880 (>1000.0)  5,927.7990 (>1000.0)       1;0      2.0770 (0.00)          5           1
test_baseline_benchmark_j1_single_working_checker       1,004,553.2550 (>1000.0)  1,004,743.6320 (>1000.0)  1,004,638.5390 (>1000.0)     71.2084 (14.88)    1,004,639.5760 (>1000.0)     91.9177 (27.21)         2;0      0.9954 (0.00)          5           1
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Legend:
  Outliers: 1 Standard Deviation from Mean; 1.5 IQR (InterQuartile Range) from 1st Quartile and 3rd Quartile.
  OPS: Operations Per Second, computed as 1 / Mean
=========================================================================================== short test summary info ===========================================================================================
FAILED tests/test_functional.py::test_functional[super_init_not_called_extensions_py310] - AssertionError: Wrong results for file "super_init_not_called_extensions_py310":
FAILED tests/test_functional.py::test_functional[typing_broken_noreturn_py372] - AssertionError: Wrong results for file "typing_broken_noreturn_py372":
FAILED tests/test_functional.py::test_functional[typing_broken_noreturn_future_import] - AssertionError: Wrong results for file "typing_broken_noreturn_future_import":
FAILED tests/test_functional.py::test_functional[typing_broken_noreturn] - AssertionError: Wrong results for file "typing_broken_noreturn":
====================================================================== 4 failed, 1649 passed, 313 skipped, 6 xfailed in 68.84s (0:01:08) ======================================================================
edreamleo@edreamleo-MS-7D09:~/ekr-pylint$
```


### Expected behavior

All unit tests pass.

### Pylint version

```shell
edreamleo@edreamleo-MS-7D09:~$ pylint --version
pylint 2.14.0-b1
astroid 2.11.5
Python 3.10.4 (main, Apr  8 2022, 17:35:13) [GCC 9.4.0]
```


### OS / Environment

edreamleo@edreamleo-MS-7D09:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.4 LTS
Release:	20.04
Codename:	focal

### Additional dependencies

edreamleo@edreamleo-MS-7D09:~$ pip3.10 freeze
apturl==0.5.2
astroid==2.11.5
attrs==21.4.0
bcrypt==3.1.7
blinker==1.4
Brlapi==0.7.0
certifi==2019.11.28
chardet==3.0.4
Click==7.0
colorama==0.4.3
command-not-found==0.3
cryptography==2.8
cupshelpers==1.0
dbus-python==1.2.16
defer==1.0.6
dill==0.3.4
distro==1.4.0
distro-info===0.23ubuntu1
docutils==0.18.1
duplicity==0.8.12.0
entrypoints==0.3
fasteners==0.14.1
future==0.18.2
gitdb==4.0.9
GitPython==3.1.27
httplib2==0.14.0
idna==2.8
iniconfig==1.1.1
isort==5.10.1
keyring==18.0.1
language-selector==0.1
launchpadlib==1.10.13
lazr.restfulclient==0.14.2
lazr.uri==1.0.3
lazy-object-proxy==1.7.1
lockfile==0.12.2
louis==3.12.0
macaroonbakery==1.3.1
Mako==1.1.0
MarkupSafe==1.1.0
mccabe==0.7.0
monotonic==1.5
netifaces==0.10.4
oauthlib==3.1.0
olefile==0.46
packaging==21.3
paramiko==2.6.0
pexpect==4.6.0
Pillow==7.0.0
platformdirs==2.5.2
pluggy==1.0.0
protobuf==3.6.1
py==1.11.0
py-cpuinfo==8.0.0
pycairo==1.16.2
pycups==1.9.73
PyGObject==3.36.0
PyJWT==1.7.1
-e git+https://github.com/PyCQA/pylint.git@2dfff17ec794c9cc08eef19b288398e67bdcb239#egg=pylint
pymacaroons==0.13.0
PyNaCl==1.3.0
pyparsing==3.0.8
PyQt6==6.3.0
PyQt6-Qt6==6.3.0
PyQt6-sip==13.3.1
PyQt6-WebEngine==6.3.0
PyQt6-WebEngine-Qt6==6.3.0
pyRFC3339==1.1
pytest==7.1.2
pytest-benchmark==3.4.1
python-apt==2.0.0+ubuntu0.20.4.7
python-dateutil==2.7.3
python-debian===0.1.36ubuntu1
pytz==2019.3
pyxdg==0.26
PyYAML==5.3.1
reportlab==3.5.34
requests==2.22.0
requests-unixsocket==0.2.0
SecretStorage==2.3.1
simplejson==3.16.0
six==1.14.0
smmap==5.0.0
systemd-python==234
tomli==2.0.1
tomlkit==0.10.2
ubuntu-advantage-tools==27.7
ubuntu-drivers-common==0.0.0
ufw==0.36
unattended-upgrades==0.1
urllib3==1.25.8
usb-creator==0.3.7
wadllib==1.3.3
wrapt==1.14.1
xkit==0.0.0

@DanielNoord I wonder if we broke the functional tests options because of ``tests/functional/s/super/super_init_not_called_extensions_py310.rc:1:0: E0015: Unrecognized option found: exclude_platforms, min_pyver (unrecognized-option)``. If `` min_pyver`` is broken some tests are not skipped when they should.

--- a/doc/development_guide/testing.rst
+++ b/doc/development_guide/testing.rst
@@ -15,7 +15,7 @@ unless they include tests.
 
 Before you start testing your code, you need to install your source-code package locally.
 Suppose you have cloned pylint into a directory, say my-pylint.
-To set up your environment for testing, open a terminal and run::
+To set up your environment for testing, open a terminal and run:
 
     cd my-pylint
     pip install -r requirements_test_min.txt

--- a/doc/development_guide/testing.rst
+++ b/doc/development_guide/testing.rst
@@ -14,9 +14,11 @@ Pylint is very well tested and has a high code coverage. New contributions are n
 unless they include tests.
 
 Before you start testing your code, you need to install your source-code package locally.
-To set up your environment for testing, open a terminal outside of your forked repository and run:
+Suppose you have cloned pylint into a directory, say my-pylint.
+To set up your environment for testing, open a terminal and run::
 
-      pip install -e <forked_repo_dir_name>
+    cd my-pylint
+    pip install -r requirements_test_min.txt
 
 This ensures your testing environment is similar to Pylint's testing environment on GitHub.
 

--- a/doc/development_guide/testing.rst
+++ b/doc/development_guide/testing.rst
@@ -14,7 +14,7 @@ Pylint is very well tested and has a high code coverage. New contributions are n
 unless they include tests.
 
 Before you start testing your code, you need to install your source-code package locally.
-Suppose you have cloned pylint into a directory, say my-pylint.
+Suppose you have cloned pylint into a directory, say ``my-pylint``.
 To set up your environment for testing, open a terminal and run:
 
     cd my-pylint

